src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -103,6 +147,12 @@ private static void initHttpClient() {
     }
 
     httpClient = clientBuilder.build();
+
+    // Monitors in a separate thread where it closes any idle connections
+    // https://hc.apache.org/httpcomponents-client-4.5.x/current/tutorial/html/connmgmt.html
+    IdleConnectionMonitorThread idleConnectionMonitorThread =
```

Comment:
where do we shut it down?

Reply:
Good point, we don't shut it down. We also never shut down the securitymanager's scheduler executor service.
I converted this thread into a deamon thread.

Reply:
> We also never shut down the securitymanager's scheduler executor service.

I don't see any reason why we shouldn't it shut it down, maybe add a fix in master?

Reply:
+1 to @sfc-gh-tzhang 's comment. if we start something explicitly we should close/shutdown

Reply:
1. I have set this thread to be daemon thread which will shutdown once the jvm shutsdown. When all of the non-daemon threads complete, daemon threads terminates automatically.
2. Or else, I will have to call shutdown() method from `SimpleIngestManager`'s `close()` method and also need to reinitialize it when a static instance of httpClient is fetched.